packages/@aws-cdk/core/README.md

```diff
@@ -1302,4 +1302,118 @@ permissions boundary attached.
 
 For more details see the [Permissions Boundary](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_iam-readme.html#permissions-boundaries) section in the IAM guide.
 
+## Policy Validation
+
+If you or your organization use any policy validation tool, such as
+[CloudFormation
+Guard](https://docs.aws.amazon.com/cfn-guard/latest/ug/what-is-guard.html) or
+[OPA](https://www.openpolicyagent.org/), to define constraints on your
+CloudFormation template, you can integrate them with the CDK at synthesis time.
+By using the appropriate plugin, you can make the CDK application check the
+generated CloudFormation template against your policies immediately after
+synthesis. If there are any violations, the synthesis will fail and a report
+will be printed to the console.
+
+### For application developers
+
+To use one or more validation plugins in your application, use the
+`validationPlugins` property of `Stage`:
+
+```ts
+// globally for the entire app (an app is a stage)
+const app = new App({
+  validationPlugins: [
+    // These hypothetical classes implement IValidationPlugin:
+    new ThirdPartyPluginX(), 
+    new ThirdPartyPluginY(),
+  ],
+});
+
+// only apply to a particular stage
+const prodStage = new Stage(app, 'ProdStage', {
+  validationPlugins: [...],
+});
+```
+
+Immediately after synthesis, all plugins registered this way will be invoked to
+validate all the templates generated in the scope you defined. In particular, if
+you register the templates in the `App` object, all templates will be subject to
+validation.
+
+By default, the report will be printed in a human readable format. If you want a
+report in JSON format, use the `synth()` method:
+
+```ts
+app.synth({
+  validationReportFormat: ValidationFormat.JSON
+});
+```
+
+### For plugin authors
+
+The communication protocol between the CDK core module and your policy tool is
+defined by the `IValidationPlugin` interface. To create a new plugin you must
+write a class that implements this interface. There are three things you need to
+implement: the plugin name (by overriding the `name` property), and the two
+methods `isReady()` and `validate()`.
+
+The method `isReady()` is called first in the workflow, to make sure that the
```

Comment:
Is this method called after synth or before? Ideally end users get early feedback that something's broken (but I also understand that might make things messy).

Reply:
Actually, this is outdated. We have removed this method.